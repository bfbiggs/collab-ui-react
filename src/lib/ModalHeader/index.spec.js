import React from 'react';
import { shallow, mount } from 'enzyme';
import { ModalHeader } from '@collab-ui/react';

describe('tests for <ModalHeader />', () => {
  it('should match SnapShot', () => {
    const container = shallow(<ModalHeader id="test" />);

    expect(container).toMatchSnapshot();
  });

  it('should render one ModalHeader', () => {
    const container = shallow(<ModalHeader />);

    expect(container.find('div').hasClass('cui-modal__header')).toEqual(true);
  });

  it('should display closable button by default', () => {
    const container = shallow(<ModalHeader />);

    expect(container.find('.cui-modal__close').length).toEqual(1);
  });

  it('should display not display closable button', () => {
    const container = shallow(<ModalHeader showCloseButton={false} />);

    expect(container.find('Icon').exists()).toEqual(false);
  });

  it('should pass onHide function to onClick', () => {
    const onClose = jest.fn();

    const container = mount(<ModalHeader onHide={onClose} />);

    container.find('button.cui-modal__close').simulate('click');
    expect(onClose).toHaveBeenCalled();
  });
});
