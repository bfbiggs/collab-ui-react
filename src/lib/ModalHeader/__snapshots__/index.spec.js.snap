// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tests for <ModalHeader /> should match SnapShot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <ModalHeader
    className=""
    headerLabel=""
    id="test"
    showCloseButton={true}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="cui-modal__title"
        >
          
        </span>,
        <CloseIcon
          className="cui-modal__close"
          onClick={[Function]}
        />,
      ],
      "className": "cui-modal__header",
      "id": "test",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "",
          "className": "cui-modal__title",
        },
        "ref": null,
        "rendered": "",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "className": "cui-modal__close",
          "onClick": [Function],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="cui-modal__title"
          >
            
          </span>,
          <CloseIcon
            className="cui-modal__close"
            onClick={[Function]}
          />,
        ],
        "className": "cui-modal__header",
        "id": "test",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "",
            "className": "cui-modal__title",
          },
          "ref": null,
          "rendered": "",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "className": "cui-modal__close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
